portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase

```diff
@@ -0,0 +1,82 @@
+definition {
+
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Remote Apps";
+	property testray.main.component.name = "Remote Apps";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			RemoteApps.tearDown();
+
+			Navigator.gotoPage(pageName = "Home");
+
+			while (IsElementPresent(locator1 = "Navigation#ANY_SITE_NAV_BUT_HOME")) {
+				var layoutName = selenium.getText("Navigation#ANY_SITE_NAV_BUT_HOME");
+
+				JSONLayout.deletePublicLayout(
+					groupName = "Guest",
+					layoutName = "${layoutName}");
+
+				Navigator.gotoPage(pageName = "Home");
+
+				PortalInstances.tearDownCP();
+			}
+		}
+	}
+
+	@description = "This is a test for LPS-164433. The JSN URL can be repeatably clicked."
+	@priority = "3"
+	test JSURLCanBeRepeated {
+		property portal.acceptance = "true";
+
+		task ("Given Remote Application > Custom Elements create page") {
+			RemoteApps.goToRemoteAppsPortlet();
+
+			RemoteApps.addCustomElement(
+				entryHtmlName = "vanilla-counter",
+				entryName = "Vanilla Counter",
+				entryURL = "http://remote-component-test.wincent.com/packages/vanilla-counter/index.js");
+		}
+
+		task ("When click repeatable button for JS URL") {
+			RemoteAppsEntry.clickRemoteEntryName(remote_name = "Vanilla Counter");
+
+			RemoteAppsEntry.addURLRow();
+
+			Type(
+				key_id = "ClientExtensionAdminPortlet_url",
+				key_index = "2",
+				locator1 = "RemoteAppsEntry#URL_ROW_INDEX",
+				value1 = "http://www.localhost.com");
+
+			PortletEntry.publish();
+		}
+
+		task ("Then new JS URL field is added") {
+			RemoteAppsEntry.clickRemoteEntryName(remote_name = "Vanilla Counter");
+
+			AssertElementPresent(
+				key_url = "http://www.localhost.com",
+				locator1 = "RemoteAppsEntry#REMOTE_URL_FIELD");
+		}
+
+		task ("And Then repeatable button + and - are present for both JS URL fields") {
```

Comment:
We need to validate 2 sets of + and - `both JS URL fields`.  This means we need xpaths for the first URL field and 2nd URL field. + and -. So 4 paths that could be simplified to 2 paths with `key_number` or something corresponding to each URL field